File: src/12장/12-4.py
N = int(input())
arr = [int(i) for i in input().split()]
 
s=int(input())
while True:
  tof=False
  for i in range(N):
    idx=i
    cmp=0
    for j in range(N-1,i,-1):
      if arr[idx]<arr[j] and j-i<=s:
        idx=j
        tof=True
        cmp=j-i
    if idx!=i:
      tmp=arr[idx]
      del arr[idx]
      arr.insert(i,tmp)
      s-=cmp
      break
  if tof==False:
    break

print(*arr)

File: src/12장/12-4.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		int n = Integer.parseInt(br.readLine());
		ArrayList<Integer> arr = new ArrayList<>();
		for (int i = 0; i < n; i++) {
			arr.add(Integer.parseInt(st.nextToken()));
		}
		int s = Integer.parseInt(br.readLine());
		while (true) {
			boolean tof = false;
			for (int i = 0; i < n - 1; i++) {
				int idx = i;
				int cmp;
				for (int j = n - 1; j >= i + 1; j--) {
					if (arr.get(idx) < arr.get(j)) {
						if (j - i <= s) {
							idx = j;
							tof = true;
							cmp = j - i;
						}
					}
				}
				if (idx != i) {
					int tmp = arr.get(idx);
					arr.remove(idx);
					arr.add(i, tmp);
					s -= cmp;
					break;
				}
			}
			if (!tof)
				break;
		}
		for (int i = 0; i < n; i++) {
			System.out.print(arr.get(i) + " ");
		}
		br.close();
	}
}